sktime/transformations/series/compose.py

```diff
@@ -96,3 +97,210 @@ def inverse_transform(self, Z, X=None):
         if not self.passthrough:
             z = self.transformer_.inverse_transform(z, X=None)
         return z
+
+
+class ColumnwiseTransformer(_SeriesToSeriesTransformer):
+    """
+    Applies a transformer for univariate series
+    columnwise to multivariate series.
+
+    Parameters
+    ----------
+    transformer : Estimator
+        scikit-learn-like or sktime-like transformer to fit and apply to series
+    columns : list of str
+            Names of columns that are supposed to be transformed
+
+     Attributes
+    ----------
+    transformers_ : dict of {str : transformer}
+        Maps columns to transformers
+
+    Example
+    -------
+    >>> from sktime.datasets import base
+    >>> from sktime.transformations.series.detrend import Detrender
+    >>> from sktime.transformations.series.compose import ColumnwiseTransformer
+
+    >>> y, X = base.load_longley()
+    >>> transformer = ColumnwiseTransformer(Detrender())
+    >>> yt = transformer.fit_transform(X)
+    """
+
+    _required_parameters = ["transformer"]
+
+    def __init__(self, transformer, columns=None):
+        self.transformer = transformer
+        self.columns = columns
```

Comment:
why do we have this here? I could find any other place where you read this. Below you only read/use `self.columns_` always

Reply:
From how I understand the scikit learn developer guide, the idea is to not change the hyperparameters in fit/in the constructor or somewhere else. This ensures that `get_params` and `set_params` don't break. We are however changing `self.columns` in case it is `None` to all the column names. This is why I think that we need `self.columns` and `self.columns_` . The developer guide also says that this should happen in fit and and not in the constructor. Therefore, I moved this logic further down to `fit` now. The guide is here: https://scikit-learn.org/stable/developers/develop.html